presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java

```diff
@@ -846,11 +843,12 @@ private boolean addNewExchangesAndBuffers()
     }
 
     @VisibleForTesting
+    @SuppressWarnings("NakedNotify")
```

Comment:
`doUpdateState` is invoked from a callback on the state machine object when the state changes